### PR TITLE
Enhance difference highlighting for keywords

### LINKED ISSUE #4682: Diffing algorithm is sometimes hard to parse
Many times diffing is super helpful but sometimes it is very hard to understand exactly what is wrong:

screen shot 2016-05-24 at 22 57 01

The diff helps because it gives me an idea of *where* is wrong but it is hard to put together exactly *what*. I have been thinking about this and what if we print the diff alongside the original lhs and rhs instead of a separate row? Something like this:

image

For maps, the lhs would show in red values that it is not on the lhs and the rhs in green values not in the lhs. Same keys different values follow the same rule. Or maybe for maps the diff: row is the best way to go still.

Thoughts?

To-do list taken from https://github.com/elixir-lang/elixir/issues/4682#issuecomment-221511114:

* [ ] 1. List diff needs improvement – make it less noisy and more content aware
* [x] 2. Handle whitespace diff in some way (thoughts?)
* [x] 3. Try separated diff

### PR DESCRIPTION
**Pull request by lexmag**
Closes #4682.

__Before:__
screen shot 2016-06-07 at 23 11 41

__After:__
screen shot 2016-06-07 at 23 10 21

### CODE REVIEW

**Comment by whatyouhide** [comment]
Great job! 🚀

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
